narayana/ArjunaJTA/jta/tests/classes/com/arjuna/ats/jta/xa/performance/JTAStoreTests.java

```diff
@@ -40,40 +40,29 @@
 @Threads(JMHConfigJTA.BT)
 @State(Scope.Benchmark)
 public class JTAStoreTests {
-    @State(Scope.Benchmark)
-	public static class BenchmarkState {
-        TransactionManager tm;
-
-		public BenchmarkState() {
-          tm = com.arjuna.ats.jta.TransactionManager.transactionManager();
-          try {
-              BeanPopulator.getDefaultInstance(CoreEnvironmentBean.class).setNodeIdentifier("0");
-          } catch (CoreEnvironmentBeanException e) {
-              throw new RuntimeException(e);
-          }
-
-          BeanPopulator.getDefaultInstance(ObjectStoreEnvironmentBean.class).
-              setObjectStoreType(VolatileStore.class.getName());
-		}
-    }
-
-    @Setup(Level.Trial)
-    public void before() {
+    static {
+      try {
+        BeanPopulator.getDefaultInstance(CoreEnvironmentBean.class).setNodeIdentifier("0");
+        BeanPopulator.getDefaultInstance(ObjectStoreEnvironmentBean.class).setObjectStoreType(VolatileStore.class.getName());
+        tm = com.arjuna.ats.jta.TransactionManager.transactionManager();
+      } catch (Exception e) {
+          throw new RuntimeException(e);
+      }
     }
 
-    @TearDown(Level.Trial)
-    public void after() {
-    }
+    private static TransactionManager tm;
+    private XAResourceImpl resource1 = new XAResourceImpl();
+    private XAResourceImpl resource2 = new XAResourceImpl();
 
     @Benchmark
-    public boolean jtaTest(BenchmarkState state) {
+    public boolean jtaTest() {
         try {
-            state.tm.begin();
+            tm.begin();
 
-            state.tm.getTransaction().enlistResource(new XAResourceImpl());
-            state.tm.getTransaction().enlistResource(new XAResourceImpl());
+            tm.getTransaction().enlistResource(resource1);
+            tm.getTransaction().enlistResource(resource2);
 
-            state.tm.commit();
+            tm.commit();
         } catch(Exception e) {
             System.err.printf("JTAStoreTests#jtaTest: %s%n", e.getMessage());
```

Comment:
There is a JMH class to consume values and avoid "dead code elimination":
http://hg.openjdk.java.net/code-tools/jmh/file/7e1c1c40872d/jmh-core/src/main/java/org/openjdk/jmh/infra/Blackhole.java